src/status_im2/contexts/wallet/receive/view.cljs

```diff
@@ -0,0 +1,96 @@
+(ns status-im2.contexts.wallet.receive.view
+  (:require
+    [quo.core :as quo]
+    [react-native.core :as rn]
+    [react-native.platform :as platform]
+    [react-native.safe-area :as safe-area]
+    [react-native.share :as share]
+    [reagent.core :as reagent]
+    [status-im2.constants :as constants]
+    [status-im2.contexts.wallet.common.sheets.network-preferences.view :as network-preferences]
+    [status-im2.contexts.wallet.common.utils :as utils]
+    [status-im2.contexts.wallet.receive.style :as style]
+    [utils.i18n :as i18n]
+    [utils.image-server :as image-server]
+    [utils.re-frame :as rf]))
+
+(def id-to-network
+  {constants/mainnet-chain-id  :ethereum
+   constants/optimism-chain-id :optimism
+   constants/arbitrum-chain-id :arbitrum})
+
+(defn- share-action
+  [address share-title]
+  (share/open
+   (if platform/ios?
+     {:activity-item-sources [{:placeholder-item {:type    "text"
+                                                  :content address}
+                               :item             {:default {:type "text"
+                                                            :content
+                                                            address}}
+                               :link-metadata    {:title share-title}}]}
+     {:title   share-title
+      :subject share-title
+      :message address})))
+
+(defn- open-preferences
+  [selected-networks]
+  (rf/dispatch [:show-bottom-sheet
+                {:content
+                 (fn []
+                   [network-preferences/view
+                    {:blur?             true
+                     :selected-networks (set @selected-networks)
+                     :on-save           (fn [chain-ids]
+                                          (rf/dispatch [:hide-bottom-sheet])
+                                          (reset! selected-networks (map #(get id-to-network %)
+                                                                         chain-ids))
+                                        )}])}]))
+
+
+(defn view
+  []
+  (let [padding-top       (:top (safe-area/get-insets))
+        wallet-type       (reagent/atom :wallet-legacy)
+        selected-networks (reagent/atom [:ethereum :optimism :arbitrum])]
```

Comment:
can we get this from a sub or from some constants some where?

Reply:
@J-Son89 design team still has not confirmed regarding the default selected networks. I can add a comment.